### ISSUE TITLE
change bounding box label index to text

### ISSUE THREAD

**Issue by hjsg1010** [ultralytics/yolov5#3578]
## ❔Question
Hi, thanks for your great research and sharing codes.

I am applying YOLOv5 to my custom dataset.

And during learning, the mosaic section on wandb.ai shows the figure below.

![image](https://user-images.githubusercontent.com/17421673/121520697-2b60c280-ca2e-11eb-9214-b1c9745be5b6.png)

However, I want to change the bbox label index to text label, such as '118' changes to 'cheesecake'.

What should I fix in code? or do I have to change my annotation?

Currently the data folder are consists of 
data_folder/  
┣ images/  
┃ ┣ train/  
┃ ┣ valid/  
┣ labels/  
┃ ┣ train/  
┃ ┣ valid/ 
┣ data_config.yaml

**Comment by glenn-jocher** [comment]
@hjsg1010 @KalenMike yes, your results are correct. The train mosaics only display the class index rather than the class name. You can see this in the Train Custom Data tutorial. We did this because the mosaics can get quite busy for high density datasets. The test dataset labels show the full name though, i.e. `test_batch0_labels.jpg`, so you may want to look at those instead.
https://docs.ultralytics.com/yolov5/tutorials/train_custom_data#local-logging

We're also developing dataset introspection tools as part of the new Ultralytics HUB. This should be ready sometime later on in summer 2021 at https://ultralytics.com

![duo3-1](https://user-images.githubusercontent.com/26833433/121523448-84acff80-c9f6-11eb-8815-aa7cd1f4caa0.png)

**Comment by hjsg1010** [comment]
> @hjsg1010 @KalenMike yes, your results are correct. The train mosaics only display the class index rather than the class name. You can see this in the Train Custom Data tutorial. We did this because the mosaics can get quite busy for high density datasets. The test dataset labels show the full name though, i.e. `test_batch0_labels.jpg`, so you may want to look at those instead.
> https://docs.ultralytics.com/yolov5/tutorials/train_custom_data#local-logging
> 
> We're also developing dataset introspection tools as part of the new Ultralytics HUB. This should be ready sometime later on in summer 2021 at https://ultralytics.com
> 
> ![duo3-1](https://user-images.githubusercontent.com/26833433/121523448-84acff80-c9f6-11eb-8815-aa7cd1f4caa0.png)

oh, Thx for quick and kind reply.
Again, thx for sharing your codes!!

**Comment by hjsg1010** [comment]
@glenn-jocher 
Hi, can I ask one more question?
I know that YOLO already does data augmentation.
However, I want to do custom data augmentation using Albumentations.
Then, where should I put augmentation pipeline code? 
.utils/dataset.py ? or train.py?

sorry for my poor English skills, and low knowledge on object detection code..

**Comment by glenn-jocher** [comment]
@hjsg1010 you can inline any Albumentations augmentations you want in the dataloader here:
https://github.com/ultralytics/yolov5/blob/095197bd4a011b867f1bb7118d1735dd84ac5ee6/utils/datasets.py#L499-L545

**Comment by hjsg1010** [comment]
> @hjsg1010 you can inline any Albumentations augmentations you want in the dataloader here:
> https://github.com/ultralytics/yolov5/blob/095197bd4a011b867f1bb7118d1735dd84ac5ee6/utils/datasets.py#L499-L545

Thx for reply. Thankyou.

**Comment by glenn-jocher** [comment]
@hjsg1010 see PR #3882 for a proposed automatic Albumentations integration.

**Comment by glenn-jocher** [comment]
@hjsg1010 good news 😃! Your original issue may now be fixed ✅ in PR #3882. This PR implements a [YOLOv5](https://github.com/ultralytics/yolov5) 🚀 + [Albumentations](https://github.com/albumentations-team/albumentations) integration. The integration will automatically apply Albumentations transforms during YOLOv5 training if `albumentations>=1.0.0` is installed in your environment.

## Get Started

To use albumentations simply `pip install -U albumentations` and then update the augmentation pipeline as you see fit in the `Albumentations` class in `yolov5/utils/augmentations.py`. Note these Albumentations operations run **in addition to** the YOLOv5 **hyperparameter** augmentations, i.e. defined in [hyp.scratch.yaml](https://github.com/ultralytics/yolov5/blob/master/data/hyps/hyp.scratch.yaml).

```python
class Albumentations:
    # YOLOv5 Albumentations class (optional, used if package is installed)
    def __init__(self):
        self.transform = None
        try:
            import albumentations as A
            check_version(A.__version__, '1.0.0')  # version requirement

            self.transform = A.Compose([
                A.Blur(p=0.1),
                A.MedianBlur(p=0.1),
                A.ToGray(p=0.01)],
                bbox_params=A.BboxParams(format='yolo', label_fields=['class_labels']))

            logging.info(colorstr('albumentations: ') + ', '.join(f'{x}' for x in self.transform.transforms))
        except ImportError:  # package not installed, skip
            pass
        except Exception as e:
            logging.info(colorstr('albumentations: ') + f'{e}')

    def __call__(self, im, labels, p=1.0):
        if self.transform and random.random() < p:
            new = self.transform(image=im, bboxes=labels[:, 1:], class_labels=labels[:, 0])  # transformed
            im, labels = new['image'], np.array([[c, *b] for c, b in zip(new['class_labels'], new['bboxes'])])
        return im, labels
```

## Example Result

Example `train_batch0.jpg` on COCO128 dataset with Blur, MedianBlur and ToGray. See the YOLOv5 [Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb) to reproduce: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

![train_batch0](https://user-images.githubusercontent.com/26833433/124400879-ff331b80-dd25-11eb-9b67-fe85ac4ca104.jpg)


# Update

To receive this YOLOv5 update:

- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload with `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
